lib/vmpooler/pool_manager.rb

```diff
@@ -1,12 +1,15 @@
 module Vmpooler
   class PoolManager
-    def initialize(config, logger, redis, graphite=nil)
+    def initialize(config, logger, redis, graphite = nil, statsd = nil)
       $config = config
 
       # Load logger library
       $logger = logger
 
-      unless graphite.nil?
+      # statsd and graphite are mutex in the context of vmpooler
+      if statsd
```

Comment:
Similarly, as above, an opportunity for refactoring; and changing the signature here if possible to not take graphite or statsd in the constructor, but internally just:

```
  def statsd
    @statsd ||= Statsd.new
  end
```